macros/parserMultiAnswer.pl

```diff
@@ -411,22 +411,21 @@ sub setMessage {
 ######################################################################
 
 #
-#  Produce the name for a named answer blank
-#  (Use the standard name for the first one, and
-#   create the prefixed names for the rest.)
+#  Produce the name for a named answer blank.
+#  (When the singleResult option is true, use the standard name for the first
+#  one, and create the prefixed names for the rest.)
 #
 sub ANS_NAME {
-  my $self = shift; my $i = shift; my $name;
+  my $self = shift; my $i = shift;
+  $self->{answerNames} = {} if !defined($self->{answerNames});
+  return $self->{answerNames}->{$i} if defined($self->{answerNames}->{$i});
   if ($self->{singleResult}) {
-    if (!$self->{id}) {
-      $name = $self->{answerName} = main::NEW_ANS_NAME();
-      $self->{id} = $answerPrefix.$name;
-    }
-    $name = $self->{id}."_".$i unless $i == 0;
+    $self->{answerNames}->{0} = main::NEW_ANS_NAME() if (!$self->{answerNames}->{0});
```

Comment:
You probably want `if !defined($self->{answerNames}{0}` (or better `unless defined($self->{answerNames}{0}`), since `!$self->{answerNames}{0}` is true in some cases when `$self->{answerNames}{0}` is defined (e.g., if it were an empty string, or the number 0).  This has been a chronic source of problems (and I am certainly guilty of doing this myself all over the place), and it is best to get in the habit of using `defined()` when that is what you really mean.

Reply:
That certainly does improve readability to use unless.  I did know the difference between defined and without, but I didn't think it mattered here.  I guess it is better though to use defined in any case here.

Reply:
Thanks for taking a look at this.